core/cont/inc/TObjArray.h

```diff
@@ -54,55 +54,55 @@ friend class TClonesArray;
+   void             Clear(Option_t *option = "") override;
    virtual void     Compress();
-   virtual void     Delete(Option_t *option="");
+   void             Delete(Option_t *option = "") override;
    virtual void     Expand(Int_t newSize);   // expand or shrink an array
-   Int_t            GetEntries() const;
+   Int_t            GetEntries() const override;
    Int_t            GetEntriesFast() const {
       return GetAbsLast() + 1;   //only OK when no gaps
    }
    Int_t            GetEntriesUnsafe() const;
-   Int_t            GetLast() const;
+   Int_t            GetLast() const override;
    TObject        **GetObjectRef() const { return fCont; };
-   TObject        **GetObjectRef(const TObject *obj) const;
-   Bool_t           IsEmpty() const { return GetAbsLast() == -1; }
-   TIterator       *MakeIterator(Bool_t dir = kIterForward) const;
-
-   void             Add(TObject *obj) { AddLast(obj); }
-   virtual void     AddFirst(TObject *obj);
-   virtual void     AddLast(TObject *obj);
-   virtual void     AddAt(TObject *obj, Int_t idx);
+   TObject **       GetObjectRef(const TObject *obj) const override;
+   Bool_t           IsEmpty() const override { return GetAbsLast() == -1; }
+   TIterator *      MakeIterator(Bool_t dir = kIterForward) const override;
+
+   void             Add(TObject *obj) override { AddLast(obj); }
+   void             AddFirst(TObject *obj) override;
+   void             AddLast(TObject *obj) override;
+   void             AddAt(TObject *obj, Int_t idx) override;
    virtual void     AddAtAndExpand(TObject *obj, Int_t idx);
    virtual Int_t    AddAtFree(TObject *obj);
-   virtual void     AddAfter(const TObject *after, TObject *obj);
-   virtual void     AddBefore(const TObject *before, TObject *obj);
-   virtual TObject *FindObject(const char *name) const;
-   virtual TObject *FindObject(const TObject *obj) const;
-   virtual TObject *RemoveAt(Int_t idx);
-   virtual TObject *Remove(TObject *obj);
+   void             AddAfter(const TObject *after, TObject *obj) override;
+   void             AddBefore(const TObject *before, TObject *obj) override;
+   TObject *        FindObject(const char *name) const override;
+   TObject *        FindObject(const TObject *obj) const override;
+   TObject *        RemoveAt(Int_t idx) override;
+   TObject *        Remove(TObject *obj) override;
    virtual void     RemoveRange(Int_t idx1, Int_t idx2);
-   virtual void     RecursiveRemove(TObject *obj);
+   void             RecursiveRemove(TObject *obj) override;
 
-   TObject         *At(Int_t idx) const;
+   TObject *         At(Int_t idx) const override;
    TObject         *UncheckedAt(Int_t i) const { return fCont[i-fLowerBound]; }
-   TObject         *Before(const TObject *obj) const;
-   TObject         *After(const TObject *obj) const;
-   TObject         *First() const;
-   TObject         *Last() const;
+   TObject *         Before(const TObject *obj) const override;
```

Comment:
If I recall correctly the coding convention says:

```suggestion
   TObject           *Before(const TObject *obj) const override;
```

Reply:
Probably, I do not remember it by heart. That's what clang-format should be for, so my IDE just formats while typing and I do not need to worry about anything.

Addressing the issue at hand, it seems like combining
```
AlignConsecutiveDeclarations: true
PointerAlignment: Right
```
does not give the expected results. It seems like clang-format needs a fix here.

Reply:
I gave it a quick try, but it seems that issue is more complicated to fix in clang-format. There is even a FIXME in the code stating:
```
  // FIXME: Currently we don't handle properly the PointerAlignment: Right
  // The * and & are not aligned and are left dangling. Something has to be done
  // about it, but it raises the question of alignment of code like:
  //   const char* const* v1;
  //   float const* v2;
  //   SomeVeryLongType const& v3;
```

Reply:
clang-format cannot be configured to replay ROOT's old coding style. The new one is supported (but is waiting for #2298 to become documented...)